Documentation/user-guides/webhook.md

```diff
@@ -1,50 +1,89 @@
 # Admission webhooks
 
-This document describes how to set up an admission webhook to validate
-PrometheusRules, and thus preventing Prometheus from loading invalid
-configuration.
+This document describes how to deploy the Prometheus operator's admission webhook service.
+
+The admission webhook can be used to:
+* Ensure that all annotations of PrometheusRule objects are coerced into string values.
+* Check that PrometheusRule objects are semantically valid.
+* Check that AlertmanagerConfig objects are semantically valid.
+* Convert AlertmanagerConfig objects between v1alpha1 and v1beta1 versions.
 
 ## Prerequisites
 
 This guide assumes that you have already [deployed the Prometheus
-Operator](getting-started.md) and that [admission
-controllers are
+Operator](getting-started.md) and that [admission controllers are
 enabled](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#how-do-i-turn-on-an-admission-controller)
 on your cluster.
 
-Admission webhooks require TLS, and as such this guide also assumes that you
-have a TLS certificate and key ready.
-
-## Preparing the Operator
+The Kubernetes API server expects admission webhooks to communicate over HTTPS
+so this guide also assumes the following:
+1. A valid TLS certificate and key has been provisioned for the admission webhook service.
+2. A secret holding the TLS certificate and key has been created.
 
-A secret needs to be created from the TLS certificate and key, assuming the
-certificate is in `tls.crt` and the key in `tls.key`:
+If you don't want to manually provision the TLS materials,
+[cert-manager](https://cert-manager.io/) is a good option.
 
-```bash
-kubectl create secret tls prometheus-operator-certs --cert=tls.crt --key=tls.key
-```
+## Admission webhook deployment
 
-The Prometheus Operator will serve the admission webhook. However, to do so, it
-requires being available over TLS, and not only plain HTTP. Thus the following
-flags need to be added to the Prometheus Operator deployment:
+Assuming that the following secret exists and contains the base64-encoded
+[PEM](https://en.wikipedia.org/wiki/Privacy-Enhanced_Mail) certificate
+(`tls.crt`) and key (`tls.key`) for the admission webhook service.
 
-* `--web.enable-tls=true` to enable the Prometheus Operator to serve its API
-  over TLS,
+```yaml
+apiVersion: v1
+data:
+  tls.crt: LS0tLS...LS0tCg==
+  tls.key: LS0tLS...LS0tCg==
+kind: Secret
+metadata:
+  name: admission-webhook-certs
+  namespace: default
+```
 
-* `--web.cert-file` to load the TLS certificate to use,
+The admission webhook's pod template should be modified to mount the secret as a
+volume and the container arguments should be modified to include the
+certificate and key:
 
-* `--web.key-file` to load the associate key.
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: prometheus-operator-admission-webhook
+  namespace: default
```

Comment:
do you think it might be a good idea to remove the `namespace` field from these examples entirely.

Reply:
yes it might make sense

Reply:
after more thinking, I realize that the webhook and conversion configurations expect the service's namespace so it's probably better to keep `default`. It's easy enough for the reader to adapt the snippets to their environment.